book/appendix/wallaroo-command-line-options.md

```diff
@@ -4,66 +4,94 @@ Every Wallaroo option exposes a set of command-line options that are used to con
-      --data/-d +[Sets address for data channel]
-      --worker-count/-w +[Sets total number of workers, including topology
-        initializer]
-      --worker-name/-n +[Sets name for this worker]
-      --metrics/-m [Sets address for external metrics (e.g. monitoring hub)]
-      --cluster-initializer/-t [Flag that sets this process as the
-        initializer]
-      --resilience-dir/-r [Sets directory to write resilience files to,
-        e.g. -r /tmp/data (no trailing slash)]
-      --ponythreads [Number of application threads. Used as part of a high-performance configuration.]
-      --ponypinasio [Used as part of a high-performance configuration.]
-      --ponynoblock [Used as part of a high-performance configuration.]
+```bash
+  --in/-i *[Comma-separated list of input addresses sources listen on]
+  --control/-c +[Sets address for initializer control channel; sets
+    control address to connect to for non-initializers]
+  --data/-d +[Sets address for initializer data channel]
+  --my-control [Optionally sets address for my data channel]
+  --my-data [Optionally sets address for my data channel]
+  --phone-home/-p [Sets address for phone home connection]
+  --external/-e [Sets address for external message channel]
+  --worker-count/-w +[Sets cluster initialier's total number of workers,
+    including cluster initializer itself]
+  --name/-n +[Sets name for this worker]
+
+  --metrics/-m [Sets address for external metrics (e.g. monitoring hub)]
+  --cluster-initializer/-t [Sets this process as the cluster
+    initializing process (that status is meaningless after init is done)]
+  --resilience-dir/-r [Sets directory to write resilience files to,
+    e.g. -r /tmp/data (no trailing slash)]
+  --log-rotation [Enables log rotation. Default: off]
+  --event-log-file-size/-l [Optionally set a file size for triggering
+    event log file rotation. If no file size is set, log rotation is only
+    triggered by external control messages sent to the address used with
+    --external]
+
+  --join/j [When a new worker is joining a running cluster, pass the
+    control channel address of any worker as the value for this
+    parameter]
+  --stop-world/u [Sets pause before state migration after the stop the
+    world]
 ```
 
 Wallaroo currently supports one input stream per pipeline. We provide comma-separated IP addresses for TCP source listeners via the `--in` parameter. Currently, the order of these addresses corresponds to the order in which pipelines are defined in your application code.
 
-Wallaroo currently supports one sink per app. Use the `--out` parameter to specify the address that this sink will write out to over TCP.
+Wallaroo currently supports one sink per application. If you are using a `TCPSInk`, use the `--out` parameter to specify the address (in a `host:port` format) that this sink will write out to over TCP. For other types of sinks, please refer to the specific sink's documentation.
```

Comment:
This shouldn't be true.  It should be that we currently support one sink per pipeline.  But there was a typo in TCPSinkConfigCLIParser that broke that.  I've opened a PR against master.  I'll talk to Andy about whether this fix should actually go to our release.  (I'm not recommending a change here right now, btw).

Reply:
@nisanharamati I have a PR which fixes that typo pending.  I think we should fix this before merging it in.  `--out` can be used to specify a list of output addresses, and just like the source, you can index into the result of calling TCPSinkConfigCLIParser to get the sink address you want.

Reply:
Actually ignore this.  Because of #1512 this section should be removed anyway.